docs: add cert import (#3523)
aws/copilot-cli: site/content/docs/developing/domain.en.md

# Domain

!!!attention
    Today, a Route 53 domain name can only be associated when running `copilot app init`.  
    If you'd like to update your application with a domain ([#3045](https://github.com/aws/copilot-cli/issues/3045)),

>     you'll need to run `copilot app delete` to remove the old one before creating a new one with `--domain` to associate a new domain.

Reviewer: Should we move this attention block after your new bullet point:
> 1. Use `--domain` when creating an application to associate a Route 53 domain in the same account.

This way they can see that there is a second option:
> 2. Use `--import-cert-arns` to bring your validated ACM certificates when creating an environment. 

and don't need to re-create the app

## Load Balanced Web Service
In Copilot, there are two ways to use custom domains for your Load Balanced Web Service:

1. Use `--domain` when creating an application to associate a Route 53 domain in the same account.
2. Use `--import-cert-arns` to bring your validated ACM certificates when creating an environment.

### Use app-associated root domain

> As mentioned in the [Application Guide](../concepts/applications.en.md#additional-app-configurations), you can configure the domain name of your app when running `copilot app init`.

Reviewer: ```suggestion
As mentioned in the [Application Guide](../concepts/applications.en.md#additional-app-configurations), you can configure the domain name of your app when running `copilot app init --domain`.
```

!!!attention
    Today, a Route 53 domain name can be associated only when running `copilot app init`.  
    If you'd like to update your application with a domain ([#3045](https://github.com/aws/copilot-cli/issues/3045)),
    you'll need to initialize a duplicate app with `--domain` and then run `copilot app delete` to

>     remove the old one. However, you can import validated ACM certificates that include your existing domains to your environment.

Reviewer 2: Now that we are back at it😂 can they really initialize a duplicate app without deleting it first?

Reviewer 2: On a different topic, what do you think of reorganizing it as something like:
```
There are two ways to use custom domains for your application:
1. `copilot app init --domain`, this will blahblahblah and Copilot will create and manage a certificate for you blahblah
2. Bring your certificate by the import feature blahblah
```

For example, `https://kudo.test.coolapp.example.aws` or `kudo-nlb.test.coolapp.example.aws:443`.

**Customized domain alias**

> If you don't like the default domain name Copilot assigns to your service, setting an [alias](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html) for your service is also very easy. You can add it directly to your [manifest's](../manifest/overview.en.md) `alias` section.

Reviewer: ```suggestion
If you don't like the default domain name Copilot assigns to your service, you can set a custom [alias](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html) for your service by editing your [manifest's](../manifest/lb-web-service.en.md#http-alias) `alias` section.
```

  alias: example-v1.aws
```

### How do I configure an alias with an existing domain?
If you want to use alias for your existing domain (not the one when you create the application), you could specify `--import-cert-arns` flag when creating the environment to import validated ACM certificates that include the alias. For example:

```
$ copilot env init --import-cert-arns arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012
```

After deploying a service to that environment, add the DNS of the Application Load Balancer (ALB) created in the environment as an A record to where your alias domain is hosted.

> 

Reviewer 2: This seems to be a less detailed explanation than the blog post - do you think it'd help if we put a link to the blog post here?

> And then after deploying the service to that environment, add the DNS of the Application Load Balancer (ALB) created in the environment as an A record to where your alias domain is hosted.

Reviewer 3: ```suggestion
After deploying a service to that environment, add the DNS of the Application Load Balancer (ALB) created in the environment as an A record to where your alias domain is hosted.
```

    - Your network load balancer's TLS listener, if the alias is for `nlb.alias` and TLS termination is enabled.
* creates an optional A record for your alias

**What does it look like?**

<iframe width="560" height="315" src="https://www.youtube.com/embed/Oyr-n59mVjI" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

### Use domain in your existing validated certificates

> If you want to use alias for your existing domain (not the one when you create the application), you could specify `--import-cert-arns` flag when creating the environment to import validated ACM certificates that include the alias. For example:

Reviewer: ```suggestion
If you own a domain outside of Route53 or for compliance reasons want to use an `alias` from your existing ACM certificates, you can specify  the `--import-cert-arns` flag when creating the environment to import validated ACM certificates that include the alias. For example:
```
